client/SMCE_Client.cpp

```diff
@@ -0,0 +1,339 @@
+    std::cout << "Creating board and compiling sketch" << std::endl;                                                                               
+    smce::Board board(exit_sketch);
+    //Compile and start board
+    compile_and_start(sketch,toolchain,board,arduino_root_dir);                                                                                      
+    std::cout << "Complete" << std::endl;                                                                                                                                                                                                                                          
+    //Create view and uart (serial) channels                                                                                     
+    auto board_view = board.view();                                                                                              
+    auto uart0 = board_view.uart_channels[0];                                                                                    
+                                                                                                                                 
+    //start listener thread for uart                                                                                             
+    std::thread uart_thread{[=] {uart_listener(uart0,file_write,arduino_root_dir);} };                                                                       
+    std::cout << "Messages received on uart from arduino is shown as "
+        << termcolor::red << "red" << termcolor::reset << " text." << std::endl;                                                                                                                             
+    // Print command menu                                                                                                        
+    print_menu();                                                                                                                                                                                                                                              
+    // Main loop, handle the command input                                                                                       
+    bool suspended = false;                                                                                                      
+    while(true){                                                                                                                 
+        std::cout << "$>";                                                                                                       
+        std::string input;                                                                                                       
+        std::getline(std::cin,input);                                                                                            
```

Comment:
this is just an idea, but perhaps put userinput on its own thread, this way you dont have users blocking your main thread :), if your board happens to crash the user would only know when they press enter..